api/generate.js

```diff
@@ -32,8 +32,15 @@ const generate = async (req, res) => {
   const mainColors = predominant.length > 4 ? predominant.slice(0, 4) : predominant;
 
   if (colors) {
+    const names = {}
     return res.json({
-      colors: mainColors.map(color => {
+      colors: mainColors.filter(color => {
+        const name = colorNamer(color).ntc[0].name
+        if(!names[name]) {
+          names[name] = true
```

Comment:
Isn't it better to just use `reduce` to create mapping for distinct colors and then use Object.values() to export that new array where there is only distinct color?

For example:

```
const colors = mainColors.reduce((filteredColors, currColor) => {
  if (filteredColor[colorNamer(color).ntc[0].name]) return filteredColors;

   const main = {
          name: colorNamer(color).ntc[0].name,
          hex: color,
          text: getTextColor(color)
        };

        const palettes = generatePalettes(main);
        
        filteredColors[colorNamer(color).ntc[0].name] =  {
          main,
          palettes,
          code: getCodes(main, palettes)
        }
         
  return filterColors
}, {})

return res.json({
  colors: Object.values(colors),
//other code
})
```

WDYT?

Reply:
You are right. It's a better solution.